File: src/Tbd.WebApi.Shared/ApiLogging/ApiLoggingMiddlewareExtensions.cs
namespace Tbd.WebApi.Shared.ApiLogging;

public static class ApiLoggingMiddlewareExtensions
{
    public static IApplicationBuilder UseApiLoggingEx(this IApplicationBuilder app) =>
        app.UseMiddleware<ApiLoggingMiddleware>();

    public static IApplicationBuilder UseApiLoggingEx(this IApplicationBuilder app, ApiLoggingOptionsModel apiLoggingOptionsModel)
    {
        if (apiLoggingOptionsModel is null)
            throw new ArgumentNullException(nameof(apiLoggingOptionsModel));

        return app.UseMiddleware<ApiLoggingMiddleware>(apiLoggingOptionsModel);
    }
}

File: src/RecipeApp.FCoreApi/Features/Extensions/ApiLoggingMiddlewareExtensions.fs
namespace RecipeApp.FCoreApi.Features.Extensions

open Microsoft.AspNetCore.Builder

open Tbd.Shared.ApiLog

open Tbd.WebApi.Shared.ApiLogging

module ApiLoggingMiddlewareExtensions =
    type IApplicationBuilder with
        member this.UseApiLoggingEx() =
            this.UseMiddleware<ApiLoggingMiddleware>() |> ignore
            this

    type IApplicationBuilder with
        member this.UseApiLoggingEx(apiLoggingOptionsModel: ApiLoggingOptionsModel) =
            if apiLoggingOptionsModel = null
            then
                nullArg "apiLoggingOptionsModel"

            this.UseMiddleware<ApiLoggingMiddleware>(apiLoggingOptionsModel) |> ignore
            this